### PR TITLE
feat(GH workflow): migrate postsubmit tests to github actions. Fixes #10747

### LINKED ISSUE #10747: Migrate postsubmit tests to GH Actions
## Description

As part of the [main tracker issue](https://github.com/kubeflow/pipelines/issues/10744), we need to migrate [postsubmit tests](https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-postsubmits.yaml) to GH Actions.

### Acceptance Criteria
* Need to be able to run locally (preferrable using KinD)
* Need to implement artifact generation ([workflow example](https://github.com/kubeflow/pipelines/blob/master/.github/workflows/backend.yml#L76), and [script example](https://github.com/kubeflow/pipelines/blob/master/scripts/deploy/github/helper-functions.sh#L252)

Impacted by this bug? Give it a 👍.

### PR DESCRIPTION
**Pull request by VDliveson**
**Description of your changes:**
Adds Github actions for postsubmit tests on push to master or release branch. Fixes #10747

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

cc @rimolive

### CODE REVIEW

File: .github/workflows/postsubmit.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    runs-on: ubuntu-latest
+    container: 
+      image: gcr.io/k8s-testimages/kubekins-e2e:v20210113-cc576af-master
+    steps:
+      - name: Create k8s Kind Cluster
+        uses: container-tools/kind-action@v2
+        with:
+          cluster_name: build-kubeflow
+      - name: Run tests
+        run: |
```
Comment (DharmitD) [inline .github/workflows/postsubmit.yml:42]: I believe you would need to add backlashes `\` at the end of each line in this command. Here's how you can format it correctly:
```
run: |
  ./test/postsubmit-tests-with-pipeline-deployment.sh \
    --kfp_deployment mkp \
    --workflow_file e2e_test_gke_v2.yaml \
    --test_result_folder e2e_test \
    --timeout "7200"
```

or you can format it to be in a single line:
```
run: ./test/postsubmit-tests-with-pipeline-deployment.sh --kfp_deployment mkp --workflow_file e2e_test_gke_v2.yaml --test_result_folder e2e_test --timeout "7200"
```

**Comment by DharmitD** [comment]
@VDliveson have you tried testing this GH Action on your fork? If yes, could you please provide the link to a successful run?

File: .github/workflows/postsubmit.yml [hunk 2]
```diff
@@ -39,7 +39,7 @@ jobs:
         with:
           cluster_name: build-kubeflow
       - name: Run tests
-        run: |
+        run: >
```
Comment (DharmitD) [inline .github/workflows/postsubmit.yml:42]: have you tested this out? I believe backslashes would be required even with the folder style.

Comment (VDliveson) [reply .github/workflows/postsubmit.yml:42]: Will update this but this still must work

Comment (VDliveson) [reply .github/workflows/postsubmit.yml:42]: https://stackoverflow.com/questions/59954185/github-action-split-long-command-into-multiple-lines

**Comment by VDliveson** [comment]
> @VDliveson have you tried testing this GH Action on your fork? If yes, could you please provide the link to a successful run?

https://github.com/VDliveson/pipelines/actions/workflows/postsubmit.yml
Running some sample tests

**Comment by DharmitD** [comment]
> @DharmitD Kindly also take a look at the periodic functional tests and is there any way to disable it. I've been spammed with 100+ mails with failed workflow. Also the old tests need python 3.7 version but we get this error on python 3.7 image
> 
> image

@VDliveson I'm looking into periodic tests GH Action and it looks like it's scheduled to run every 5 minutes, that's the problem. I will bring this up in community but make sure to turn off email notifications for repo GH Actions so it doesn't incessantly ping you with so many notifications.

**Comment by DharmitD** [comment]
@chensun looks like the postsubmit prow config forwards test results to TestGrid: https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-postsubmits.yaml#L14
Do we utilize this notifications’ mechanism? Do we need it in the kind workflows too?

**Comment by VDliveson** [comment]
@DharmitD Any updates on this ?

**Pull request closed without merging**